src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -15,15 +17,26 @@ namespace NuGet.Services.EndToEnd.Support
 {
     public class V2V3SearchClient
     {
+        /// <summary>
+        /// In case cloud service information is not provided, this will be the fallback instance count.
+        /// </summary>
+        private const int DefaultServiceInstanceCount = 2;
```

Comment:
> The first case will never happened - if the instance count is less then 2 we have a serious problem with service config that needs to be fixed ASAP.

Okay, sounds good.

> Don't understand when second case is possible. Point me to the test that will fail.

Suppose you are running against a search service with 3 or more instances but you have defaulted to 2 in the test configuration.

See:
https://github.com/NuGet/NuGet.Services.EndToEnd/blob/9af5255d0610a97664f8feae5acca155c04010cd/src/NuGet.Services.EndToEnd/PushTests.cs#L53

This line should only pass if the package is available on _all_ search instances. However, if you have 2 configured and 3 in reality the aforementioned could pass when that third instance does not yet have the package.

Then this line could fail if the load balancer sends the request to that third instance:
https://github.com/NuGet/NuGet.Services.EndToEnd/blob/9af5255d0610a97664f8feae5acca155c04010cd/src/NuGet.Services.EndToEnd/PushTests.cs#L66